.github/workflows/testing.yml

```diff
@@ -0,0 +1,63 @@
+name: GitHub Actions Linux Testing
+
+on:
+  push:
+    branches:
+    - master
+    - 'v1.*'
+  pull_request:
+  schedule:
+  - cron: 54 19 * * SUN # weekly at a "random" time
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        jre: [8, 11]
+      fail-fast: false # Should swap to true if we grow a large matrix
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v2
+      with:
+        java-version: ${{ matrix.jre }}
+        distribution: 'adopt'
+
+    - name: Gradle cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
```

Comment:
Do we need to have separate caches per jre versions? https://docs.gradle.org/current/userguide/common_caching_problems.html#java_version_tracking

Reply:
No. That's a different type of caching. That is for "build caching" vs "artifact caching" which we are messing with here.